gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,248 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+_logger = get_logger_adapter(__name__)
+
+
+_PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")
+
+
+def _is_python_m_proc(process: Process) -> bool:
+    """
+    Checks whether the process ran as "python -m ..." pattern.
+    """
+    args = process.cmdline()
+
+    # Just to be on the safe side, module name must be run after the "-m" argument, python enforces it.
+    if len(args) < 3:
+        return False
+
+    return _is_python_bin(args[0]) and args[1] == "-m"
+
+
+def _is_python_bin(bin_name: str) -> bool:
+    return _PYTHON_BIN_RE.match(os.path.basename(bin_name)) is not None
+
+
+class _ApplicationIdentifier(metaclass=ABCMeta):
+    @abstractmethod
+    def get_application_name(self, process: Process) -> Optional[str]:
+        pass
+
+
+_IP_PORT_RE = re.compile(r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(?:\:\d{2,5})?$")  # Matches against (ip(:port))
+_NON_AVAILABLE_ARG = str()
+
+
+def _get_cli_arg_by_name(args: List[str], arg_name: str, check_for_equals_arg: bool = False) -> str:
+    if arg_name in args:
+        return args[args.index(arg_name) + 1]
+
+    if check_for_equals_arg:
+        for arg in args:
+            arg_key, _, arg_val = arg.rpartition("=")
+            if arg_key == arg_name:
+                return arg_val
+
+    return _NON_AVAILABLE_ARG
+
+
+def _get_cli_arg_by_index(args: List[str], index: int) -> str:
+    try:
+        return args[index]
+    except IndexError:
+        return _NON_AVAILABLE_ARG
+
+
+def _append_python_module_to_proc_wd(process: Process, module_path: str) -> str:
+    if os.path.isabs(module_path):
+        return module_path
+
+    return os.path.join(process.cwd(), module_path)
```

Comment:
Maybe use `realpath`? So it'll look like this:
```python
In [4]: os.path.realpath(os.path.join('/abs/path', './python_app.py'))
Out[4]: '/abs/path/python_app.py'
```
As opposed to like this:
```python
In [5]: os.path.join('/abs/path', './python_app.py')
Out[5]: '/abs/path/./python_app.py'
```

(Also in the `isabs` case)